lib/runtime/version.go

```diff
@@ -30,9 +30,24 @@ type Version struct {
 }
 
 var (
+	// Blake2-8("TaggedTransactionQueue")
+	encodedTaggedTransactionQueue = [8]byte{0xd2, 0xbc, 0x98, 0x97, 0xee, 0xd0, 0x8f, 0x15}
+
 	ErrDecodingVersionField = errors.New("decoding version field")
 )
 
+// TaggedTransactionQueueVersion returns the TaggedTransactionQueueAPI version
+func (*Version) TaggedTransactionQueueVersion(runtimeVersion Version) uint32 {
```

Comment:
What happens if the version returned is not supported by Gossamer?
You might need to handle it when the wasmer instance sets the Version field (`NewInstance` and `HandleRuntimeChanges` I think), and return an error.

Reply:
We currently handle it in transaction construction, where an error is returned if this happens (for both new and legacy unsupported versions)

Reply:
I'm not sure why we would globally need to track supported versions. Unless I'm missing something I feel like we can just check version where we actually construct transactions

Reply:
As in, the transaction queue version given by the runtime to Gossamer has to be supported by Gossamer (like the v1 state trie) right? If so we should enforce to error out if the runtime gives a version not implemented yet.